staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -445,6 +458,71 @@ func (rt *debuggingRoundTripper) RoundTrip(req *http.Request) (*http.Response, e
 	}
 
 	startTime := time.Now()
+
+	if rt.levels[DebugDetailedTiming] {
+		var getConn, dnsStart, dialStart, tlsStart, serverStart time.Time
+		var host string
+		trace := &httptrace.ClientTrace{
+			// DNS
+			DNSStart: func(info httptrace.DNSStartInfo) {
+				dnsStart = time.Now()
```

Comment:
is it ever possible for this to be called multiple times? for example, if the request returns a redirect that is followed to a different dns name?

Reply:
it would be simpler to reason about to always `reqInfo.muTrace.Lock()` / `defer reqInfo.muTrace.Unlock()` in these trace methods. is there a reason not to do that?

Reply:
Traces doesn't span multiple redirects
https://github.com/golang/go/commit/864859d209a144eb1a703381ec96b9c897d4e493

I was following the godocs of the trace, but since this is for debugging with verbosity 9, I think we can use the bruteforce approach if you want

Reply:
added lock, unlock per method that writes on reqInfo